lighthouse-core/audits/dobetterweb/uses-http2.js

```diff
@@ -43,67 +58,174 @@ class UsesHTTP2Audit extends Audit {
+
+    // Update all the protocols to reflect implementing our recommendations
+    /** @type {Map<string, string>} */
+    const originalProtocols = new Map();
+    graph.traverse(node => {
+      if (node.type !== 'network') return;
+      if (!urlsToChange.has(node.record.url)) return;
+
+      originalProtocols.set(node.record.requestId, node.record.protocol);
+      node.record.protocol = 'h2';
+    });
+
+    const simulationAfter = simulator.simulate(graph, {label: afterLabel, flexibleOrdering});
+
+    // Restore the original protocol after we've done our simulation
+    graph.traverse(node => {
+      if (node.type !== 'network') return;
+      const originalProtocol = originalProtocols.get(node.record.requestId);
+      if (originalProtocol === undefined) return;
+      node.record.protocol = originalProtocol;
+    });
+
+    const savingsOnOverallLoad = simulationBefore.timeInMs - simulationAfter.timeInMs;
+    const savingsOnTTI = Interactive.getLastLongTaskEndTime(simulationBefore.nodeTimings) -
+      Interactive.getLastLongTaskEndTime(simulationAfter.nodeTimings);
+    const savings = Math.max(savingsOnTTI, savingsOnOverallLoad);
+
+    // Round waste to nearest 10ms
+    return Math.round(Math.max(savings, 0) / 10) * 10;
+  }
+
+  /**
+   * Determines whether a network request is a "static resource" that would benefit from H2 multiplexing.
+   * XHRs, tracking pixels, etc generally don't benefit as much because they aren't requested en-masse
+   * for the same origin at the exact same time.
+   *
+   * @param {LH.Artifacts.NetworkRequest} networkRequest
+   * @return {boolean}
+   */
+  static isStaticAsset(networkRequest) {
+    if (!STATIC_RESOURCE_TYPES.has(networkRequest.resourceType)) return false;
+
+    // Resources from third-parties that are less than 100 bytes are usually tracking pixels, not actual resources.
+    // They can masquerade as static types though (gifs, documents, etc)
+    if (networkRequest.resourceSize < 100) {
+      const entity = ThirdPartyAudit.getEntitySafe(networkRequest.url);
+      if (entity) return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Determine the set of resources that aren't HTTP/2 but should be.
+   * We're a little conservative about what we surface for a few reasons:
+   *
+   *    - The simulator approximation of HTTP/2 is a little more generous than reality.
+   *    - There's a bit of debate surrounding HTTP/2 due to its worse performance in environments with high packet loss.
+   *    - It's something that you'd have absolutely zero control over with a third-party (can't defer to fix it for example).
+   *
+   * Therefore, we only surface requests that were...
+   *
+   *    - Served over HTTP/1.1 or earlier
+   *    - Served over an origin that serves at least 6 static asset requests
+   *      (if there aren't more requests than browser's max/host, multiplexing isn't as big a deal)
```

Comment:
So does `HTTP/1` share a connection with up to 6 requests? I thought it was one request = one connection but I def. need to read up on this more.

Reply:
No you're right it's `one request = one connection` but the point of this check is that if we could've already downloaded all the resources in parallel by making separate connections, we're not really gaining much with multiplexing in HTTP/2.